Hey guys, sorry for the late reply, I was out for a while.

It's very peculiar that you are building it successfully with the normal `stdenv`. I get the following error when trying:

```
these derivations will be built:
  /nix/store/znklyd38kk4qr7nsv17ikpi8c9ppnlks-source.drv
  /nix/store/h92i29wsfl5fzg9hj3inxh8chyaizwwc-cpp-utilities-5.0.0.drv
these paths will be fetched (4.56 MiB download, 26.71 MiB unpacked):
  /nix/store/97g3zdlrz7jr11w6p13w6l4dj2qcxjm3-cmake-3.15.1
  /nix/store/f15rdzsyhvif2v7hmcxs6mrg3syihnki-rhash-1.3.8
  /nix/store/z206871lsq3sqs6szsbsld8n9cqhf435-cppunit-1.14.0
copying path '/nix/store/z206871lsq3sqs6szsbsld8n9cqhf435-cppunit-1.14.0' from 'https://cache.nixos.org'...
copying path '/nix/store/f15rdzsyhvif2v7hmcxs6mrg3syihnki-rhash-1.3.8' from 'https://cache.nixos.org'...
building '/nix/store/znklyd38kk4qr7nsv17ikpi8c9ppnlks-source.drv'...
copying path '/nix/store/97g3zdlrz7jr11w6p13w6l4dj2qcxjm3-cmake-3.15.1' from 'https://cache.nixos.org'...

trying https://github.com/Martchus/cpp-utilities/archive/v5.0.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   130    0   130    0     0    293      0 --:--:-- --:--:-- --:--:--   293
100  148k    0  148k    0     0  97862      0 --:--:--  0:00:01 --:--:--  138k
unpacking source archive /build/v5.0.0.tar.gz
building '/nix/store/h92i29wsfl5fzg9hj3inxh8chyaizwwc-cpp-utilities-5.0.0.drv'...
unpacking sources
unpacking source archive /nix/store/c4m5rgpsmh4bhlqmvkwspzgq7rg30zjy-source
source root is source
patching sources
configuring
fixing cmake files...
cmake flags: -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DCMAKE_INSTALL_LOCALEDIR=/nix/store/afapac79cyy4i3zxx1h7s8wh8xd05a45-cpp-utilities-5.0.0/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/afapac79cyy4i3zxx1h7s8wh8xd05a45-cpp-utilities-5.0.0/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/afapac79cyy4i3zxx1h7s8wh8xd05a45-cpp-utilities-5.0.0/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/afapac79cyy4i3zxx1h7s8wh8xd05a45-cpp-utilities-5.0.0/share/doc/ -DCMAKE_INSTALL_INFODIR=/nix/store/afapac79cyy4i3zxx1h7s8wh8xd05a45-cpp-utilities-5.0.0/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/afapac79cyy4i3zxx1h7s8wh8xd05a45-cpp-utilities-5.0.0/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/afapac79cyy4i3zxx1h7s8wh8xd05a45-cpp-utilities-5.0.0/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/afapac79cyy4i3zxx1h7s8wh8xd05a45-cpp-utilities-5.0.0/include -DCMAKE_INSTALL_SBINDIR=/nix/store/afapac79cyy4i3zxx1h7s8wh8xd05a45-cpp-utilities-5.0.0/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/afapac79cyy4i3zxx1h7s8wh8xd05a45-cpp-utilities-5.0.0/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/afapac79cyy4i3zxx1h7s8wh8xd05a45-cpp-utilities-5.0.0/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_DEPLOYMENT_TARGET= -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=last -DCMAKE_STRIP=/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/strip -DCMAKE_RANLIB=/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ranlib -DCMAKE_AR=/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/afapac79cyy4i3zxx1h7s8wh8xd05a45-cpp-utilities-5.0.0
-- The C compiler identification is GNU 7.4.0
-- The CXX compiler identification is GNU 7.4.0
-- Check for working C compiler: /nix/store/382vgzzm7xrkqfply7y1h3fn03hp68m5-gcc-wrapper-7.4.0/bin/gcc
-- Check for working C compiler: /nix/store/382vgzzm7xrkqfply7y1h3fn03hp68m5-gcc-wrapper-7.4.0/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /nix/store/382vgzzm7xrkqfply7y1h3fn03hp68m5-gcc-wrapper-7.4.0/bin/g++
-- Check for working CXX compiler: /nix/store/382vgzzm7xrkqfply7y1h3fn03hp68m5-gcc-wrapper-7.4.0/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for iconv
-- Looking for iconv - found
-- Using iconv from the standard library for target c++utilities.
-- Using __gnu_cxx::stdio_filebuf for NativeFileStream
-- Performing Test COULD_COMPILE_TEST_PROGRAM_0
-- Performing Test COULD_COMPILE_TEST_PROGRAM_0 - Failed
-- Performing Test COULD_COMPILE_TEST_PROGRAM_1
-- Performing Test COULD_COMPILE_TEST_PROGRAM_1 - Failed
-- Performing Test COULD_COMPILE_TEST_PROGRAM_2
-- Performing Test COULD_COMPILE_TEST_PROGRAM_2 - Failed
CMake Error at cmake/modules/3rdParty.cmake:304 (message):
  Unable to compile a simple std::filesystem example.  A compiler supporting
  C++17 is required to build this project.
Call Stack (most recent call first):
  CMakeLists.txt:171 (use_standard_filesystem)


-- Configuring incomplete, errors occurred!
See also "/build/source/build/CMakeFiles/CMakeOutput.log".
See also "/build/source/build/CMakeFiles/CMakeError.log".
builder for '/nix/store/h92i29wsfl5fzg9hj3inxh8chyaizwwc-cpp-utilities-5.0.0.drv' failed with exit code 1
error: build of '/nix/store/h92i29wsfl5fzg9hj3inxh8chyaizwwc-cpp-utilities-5.0.0.drv' failed
```

--- a/pkgs/development/libraries/cpp-utilities/default.nix
+++ b/pkgs/development/libraries/cpp-utilities/default.nix
@@ -2,13 +2,13 @@
 
 stdenv.mkDerivation rec {
   pname = "cpp-utilities";
-  version = "4.17.1";
+  version = "5.0.0";
 
   src = fetchFromGitHub {
     owner = "Martchus";
     repo = pname;
     rev = "v${version}";
-    sha256 = "12088cwg3jbqipmbn4843w1cgxi1q6vwx47gy042rkfvbk6azhxl";
+    sha256 = "031kiawmrp40sqgd9g5qd82ljglcm9xvynicsn4n8n4ngfgbnc0m";
   };
 
   nativeBuildInputs = [ cmake ];

